cloudferrylib/os/network/neutron.py

```diff
@@ -106,6 +107,44 @@ def read_info(self, **kwargs):
             info['lb_vips'] = self.get_lb_vips()
         return info
 
+    def show_quota(self, tenant_id=''):
+        return self.neutron_client.show_quota(tenant_id)
+
+    def list_quotas(self):
+        return self.neutron_client.list_quotas()['quotas']
+
+    def get_quota(self, tenant_id):
+        # return structure {'name_tenant': {'subnet': 10, ...}, ...}
+        tenants = {}
+        if not tenant_id:
+            tenants_obj = self.identity_client.get_tenants_list()
+            tenants = {t.id: t.name for t in tenants_obj}
+        else:
+            tenants[tenant_id] = self.identity_client.\
+                try_get_tenant_name_by_id(tenant_id)
+        data = {
+        }
+        if self.config.network.get_all_quota:
+            for t_id, t_val in tenants.iteritems():
+                data[t_val] = self.neutron_client.show_quota(t_id)
+        else:
+            for t in self.neutron_client.list_quotas()['quotas']:
+                if (not tenant_id) or (tenant_id == t['tenant_id']):
+                    tenant_name = self.identity_client.\
+                        try_get_tenant_name_by_id(t['tenant_id'])
+                    del t['tenant_id']
```

Comment:
`del` is a sign of poor design, please refactor. It seems `del` can be omitted by creating a separate object with all required information in `data[tenant_name]`.